### PR TITLE
fix: checksum.py for PayTM

### LINKED ISSUE #6609: payTM is not functional
**Describe the bug**

payTM is not working.
**To Reproduce**
1. Activate payTM with `Test & Development` Option.
2. Make an order using payTM
3. Click on `pay with payTM`

**Expected behavior**

The button should function as expected.
**Stacktrace**

```py
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 70, in sentry_patched_wsgi_app
    environ, start_response
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 120, in __call__
    reraise(*_capture_exception(hub))
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 57, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 116, in __call__
    _sentry_start_response, start_response, span
  File "/usr/local/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 69, in <lambda>
    return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/kush/open-event-server/app/__init__.py", line 77, in __call__
    return self.app(environ, start_response)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/kush/open-event-server/app/api/helpers/permissions.py", line 45, in decorator
    return fn(*args, **kwargs)
  File "/Users/kush/open-event-server/app/api/orders.py", line 659, in initiate_transaction
    checksum = PaytmPaymentsManager.generate_checksum(paytm_params)
  File "/Users/kush/open-event-server/app/api/helpers/payment.py", line 342, in generate_checksum
    return checksum.generate_checksum_by_str(json.dumps(paytm_params["body"]), merchant_key)
  File "/Users/kush/open-event-server/app/api/helpers/checksum.py", line 55, in generate_checksum_by_str
    return __encode__(hash_string, IV, merchant_key)
  File "/Users/kush/open-event-server/app/api/helpers/checksum.py", line 105, in __encode__
    c = AES.new(key, AES.MODE_CBC, iv)
  File "/usr/local/lib/python3.7/site-packages/Crypto/Cipher/AES.py", line 232, in new
    return _create_cipher(sys.modules[__name__], key, mode, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/Crypto/Cipher/__init__.py", line 79, in _create_cipher
    return modes[mode](factory, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/Crypto/Cipher/_mode_cbc.py", line 274, in _create_cbc_cipher
    cipher_state = factory._create_base_cipher(kwargs)
  File "/usr/local/lib/python3.7/site-packages/Crypto/Cipher/AES.py", line 103, in _create_base_cipher
    result = start_operation(c_uint8_ptr(key),
  File "/usr/local/lib/python3.7/site-packages/Crypto/Util/_raw_api.py", line 144, in c_uint8_ptr
    raise TypeError("Object type %s cannot be passed to C code" % type(data))
TypeError: Object type <class 'str'> cannot be passed to C code
```

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #6609

### CODE REVIEW

File: app/api/helpers/checksum.py [hunk 1]
```diff
@@ -97,13 +97,17 @@ def __pad__(string_literal):
 def __unpad__(string_literal):
     return string_literal[0:-ord(string_literal[-1])]
 
-
 def __encode__(to_encode, iv, key):
+
+    # to_encode = to_encode.encode("UTF-8")
+    # iv = iv.encode("UTF-8")
+    # key = key.encode("UTF-8")
+
```
Comment (iamareebjamal) [inline app/api/helpers/checksum.py]: Remove

**Comment by kushthedude** [comment]
> @kushthedude Are you sure that this fixes the C code error? What's the reason for the encoding?

@mrsaicharan1 https://stackoverflow.com/a/7585619/9760446

File: app/api/helpers/checksum.py (hunk 1 above)
Comment (kushthedude) [reply app/api/helpers/checksum.py]: I thought to leave comments for below reference, Hence removing them would be a better option than converting them here ?

Comment (iamareebjamal) [reply app/api/helpers/checksum.py]: You're repeating the encoding again and again, why not do it once and be done with it

Comment (kushthedude) [reply app/api/helpers/checksum.py]: These three variables are used in three different functions occurring once, Hence changing them inside one scope won't work. That's why doing at the same time.

Comment (iamareebjamal) [reply app/api/helpers/checksum.py]: OK. Then remove the comment

**Comment by kushthedude** [comment]
@iamareebjamal Please check

**Comment by mrsaicharan1** [comment]
@kushthedude This checksum file was taken from the official payTM repo as the checksum logic. Try to raise the issue there too if you're sure it's fixed here.

**Comment by kushthedude** [comment]
One of the maintainer has already raised the issue and this was the
solution provided there. Also the payTM repo is not that active.

**Comment by mrsaicharan1** [comment]
> One of the maintainer has already raised the issue and this was the solution provided there. Also the payTM repo is not that active.

Link to that issue?

**Comment by kushthedude** [comment]
Refer this
https://github.com/Paytm-Payments/Paytm_App_Checksum_Kit_Python/issues/4
and the stackoverflow answer I have mentioned.

File: app/api/helpers/checksum.py [hunk 2]
```diff
@@ -97,13 +97,13 @@ def __pad__(string_literal):
 def __unpad__(string_literal):
     return string_literal[0:-ord(string_literal[-1])]
 
-
 def __encode__(to_encode, iv, key):
```
Comment (iamareebjamal) [inline app/api/helpers/checksum.py]:
```suggestion

def __encode__(to_encode, iv, key):
```

File: app/api/helpers/checksum.py [hunk 3]
```diff
@@ -99,11 +99,12 @@ def __unpad__(string_literal):
 
 
 def __encode__(to_encode, iv, key):
+
```
Comment (iamareebjamal) [inline app/api/helpers/checksum.py]:
```suggestion
```

**Pull request merged**